src/dataecon/C.jl

```diff
@@ -0,0 +1,292 @@
+module C
+
+using DataEcon_jll
+export DataEcon_jll
+
+using CEnum
+
+function de_error(msg, len)
+    ccall((:de_error, libdaec), Cint, (Ptr{Cchar}, Csize_t), msg, len)
+end
+
+function de_error_source(msg, len)
+    ccall((:de_error_source, libdaec), Cint, (Ptr{Cchar}, Csize_t), msg, len)
+end
+
+function de_clear_error()
+    ccall((:de_clear_error, libdaec), Cint, ())
+end
+
+@cenum var"##Ctag#328"::Int32 begin
+    DE_SUCCESS = 0
+    DE_ERR_ALLOC = -1000
+    DE_BAD_AXIS_TYPE = -999
+    DE_BAD_CLASS = -998
+    DE_BAD_TYPE = -997
+    DE_BAD_NAME = -996
+    DE_SHORT_BUF = -995
+    DE_OBJ_DNE = -994
+    DE_AXIS_DNE = -993
+    DE_ARG = -992
+    DE_NO_OBJ = -991
+    DE_EXISTS = -990
+    DE_BAD_OBJ = -989
+    DE_NULL = -988
+    DE_DEL_ROOT = -987
+    DE_MIS_ATTR = -986
+    DE_INTERNAL = -985
+end
+
+const de_file = Ptr{Cvoid}
+
+function de_open(fname, de)
+    ccall((:de_open, libdaec), Cint, (Ptr{Cchar}, Ptr{de_file}), fname, de)
+end
+
+function de_close(de)
+    ccall((:de_close, libdaec), Cint, (de_file,), de)
+end
+
+@cenum class_t::Int32 begin
+    class_catalog = 0
+    class_scalar = 1
+    class_vector = 2
+    class_tseries = 2
+    class_matrix = 3
+    class_mvtseries = 3
+    class_any = -1
+end
+
+@cenum type_t::Int32 begin
+    type_none = 0
+    type_integer = 1
+    type_signed = 1
+    type_unsigned = 2
+    type_date = 3
+    type_float = 4
+    type_complex = 5
+    type_string = 6
+    type_other_scalar = 7
+    type_vector = 10
+    type_range = 11
+    type_tseries = 12
+    type_other_1d = 13
+    type_matrix = 20
+    type_mvtseries = 21
+    type_other_2d = 22
+    type_any = -1
+end
+
+@cenum frequency_t::UInt32 begin
+    freq_none = 0
+    freq_unit = 1
+    freq_daily = 4
+    freq_bdaily = 5
+    freq_monthly = 8
+    freq_weekly = 16
+    freq_weekly_sun = 16
+    freq_weekly_mon = 17
+    freq_weekly_tue = 18
+    freq_weekly_wed = 19
+    freq_weekly_thu = 20
+    freq_weekly_fri = 21
+    freq_weekly_sat = 22
+    freq_quarterly = 32
```

Comment:
I was thinking that one can do arithmetic with the frequency offsets.  For example, Jan is the first month of the quarter, so `freq_quarterly_jan = freq_quarterly + 1`.  And the same goes for `freq_halfyearly`, and `freq_yearly`.  

The same also goes for `freq_weekly`, except that now it's day of the week, rather than month of quarter/half-year/year. Now, the question is, what is the offset of Sunday? I think it can be 0 or 7. How do we pick one vs the other? Or do we allow both? 

I picked 0 because I . . . I guess didn't think much about it.  

In fact, we're not really using this. If you look at "I.jl:83"
```
_to_de_scalar_freq(::Type{Weekly{end_day}}) where {end_day} = C.frequency_t(C.freq_weekly + mod1(end_day, 7))
```
So, our weekly-Sunday, which is `Weekly{7}`, is stored as `freq_weekly + 7`, which isn't a named constant in C 😆 .  But it's still an integer. It works because when loading data we interpret that same integer as the same `Weekly{7}` frequency that we started with.

Reply:
I've added `freq_weekly_sun0` and `freq_weekly_sun7` to cover both cases. Also `freq_weekly_sun` is now an alias for `freq_weekly_sun7`

Reply:
I see, I like this arithmetic idea.